crates/rpc-server/src/registry.rs

```diff
@@ -131,14 +135,30 @@ async fn ping() -> Result<String> {
 async fn get_block(
     Params((block_hash,)): Params<(JsonH256,)>,
     store: Data<Store>,
-) -> Result<Option<L2BlockView>> {
+    rollup_config: Data<RollupConfig>,
+) -> Result<Option<L2BlockWithStatus>> {
     let block_hash = to_h256(block_hash);
     let db = store.begin_transaction();
-    let block_opt = db.get_block(&block_hash)?.map(|block| {
-        let block_view: L2BlockView = block.into();
-        block_view
-    });
-    Ok(block_opt)
+    let block_opt = db.get_block(&block_hash)?;
+    let mut status = L2BlockStatus::Unfinalized;
+    if let Some(ref block) = block_opt {
+        if db.block_smt()?.get(&block.smt_key().into())?.is_zero() {
+            return Ok(None);
```

Comment:
If a block is reverted, we can't find it from block_smt, which means this RPC would never return a reverted block.

Reply:
Block's smt key is block number. For reverted block, we should check reverted block smt first. If valid tip block number is smaller than queried reverted block, block smt will return H256::zero. If valid tip block number is equal or bigger than reverted block, block smt will return different block hash.